lib/puppet/util/run_mode.rb

```diff
@@ -85,21 +85,21 @@ def code_dir
       end
 
       def var_dir
-        which_dir(File.join(windows_common_base("var")), "~/.puppet/var")
+        which_dir(File.join(windows_common_base("puppet/cache")), "~/.puppet/var")
```

Comment:
yes it will need to create both `cache` and `var`. The `:logdir` and `:rundir` will be created by puppet as a result of it being directory settings.